src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -196,8 +197,11 @@ protected void purchase(final boolean purchaseForBid, int PUsToSpend, final IPur
 
       // Check if any place territories exist
       final Map<Territory, ProPurchaseTerritory> purchaseTerritories = purchaseUtils.findPurchaseTerritories(player);
-      if (purchaseTerritories.isEmpty()) {
-        ProLogger.info("No possible place territories owned so exiting purchase logic");
+      final List<Territory> possibleFactoryTerritories =
```

Comment:
This is a difficult change to understand because it appears that the method `territoryHasNoInfraFactoryAndIsNotConqueredOwnedLand` is misnamed. "No Infra Factory" sounds like there would be no infrastructure factories. Yet the context and usage seems otherwise. Looking at the source code of the method in question, it looks like it does find things that have a factory. I strongly recommend we rename the method to avoid further confusion about what the method does, otherwise the name is contradictory to the function. Unless I'm mistake about what the method does it should be named  (note the dropped "no"): "territoryHasInfraFactoryAndIsNotConqueredOwnedLand"

Second, perhaps a v2.0 change, you've a long method code smell.  purchaseTerritories is used only locally for the if check. Extracting these lines into a single method that returns the boolean value needed for the if check would help clean things up.